ota-plus-web/app/com/advancedtelematic/auth/AuthenticatedAction.scala

```diff
@@ -72,29 +74,43 @@ abstract class SecuredAction[R[_] <: AuthorizedRequest[_]] extends ActionBuilder
 
   val log = Logger(this.getClass)
 
-  def invokeBlock[A](request: Request[A], block: (R[A]) => Future[Result]): Future[Result] = {
+  def invokeBlock[A](request: Request[A], block: R[A] => Future[Result]): Future[Result] = {
     authRequest(request) match {
       case Success(x) =>
         tokenVerification(x.accessToken)
           .flatMap(isValid => if (isValid) block(x) else Future.successful(unauthorizedResult))
+          .recover {
+            case UnexpectedResponse(r) if r.status == UNAUTHORIZED_401 => unauthorizedResult
```

Comment:
Why do not redirect to login page?

Reply:
I think that depends on each implementation of `SecuredAction[_]`. Some want to redirect to the login and some want to return "Forbidden". As it happens only `IdentityAction` should end up in this recover, but I think it makes sense for consistency to let each implementation decide.